src/routes/delegates/delegates.controller.ts

```diff
@@ -91,4 +92,22 @@ export class DelegatesController {
       deleteDelegateDto,
     );
   }
+
+  @Delete('chains/:chainId/safes/:safeAddress/delegates/:delegateAddress')
+  async deleteSafeDelegate(
+    @Param('chainId') chainId: string,
+    @Param('safeAddress') safeAddress: string,
+    @Param('delegateAddress') delegateAddress: string,
+    @Body() deleteSafeDelegateRequest: DeleteSafeDelegateDto,
+  ): Promise<unknown> {
+    if (safeAddress !== deleteSafeDelegateRequest.safe) {
+      throw new BadRequestException();
+    }
```

Comment:
Is this a new check (compared to the old CGW)?

Reply:
Yes, this is a new check, sorry, I forgot to comment on this explicitly in the PR.

In the current CGW specification, both the Safe address and the delegate address are present in two places: in the URL and in the body request. Since we are sending them also in two places (see [the network call in TransactionApiService](https://github.com/5afe/safe-client-gateway-nest/blob/2953e1e713197878f94a45bc13efc0590fe68cfd/src/datasources/transaction-api/transaction-api.service.ts#L224)), I think it could be a good practice to validate these values are equal between URL and request body. Do you think this validation check is excessive? (I can't foresee a scenario where the clients may want to have different values between the URL and request body)

Reply:
My concern is that this might change what the clients are expecting. Did you check with the clients this behaviour?

My suggestion is to keep it out for now but keep it in mind (or even have an issue) to evaluate adding these checks. It might be worth reviewing in other routes too with the clients.

Reply:
Ok, maybe it's better to remove these validation checks. I didn't check with the clients yet but having an issue reviewing this sounds good. Then we can include these kinds of checks, or even check if having the same data in several parts of the requests make sense at all.

Reply:
Removed in [74dde61](https://github.com/5afe/safe-client-gateway-nest/pull/270/commits/74dde610ba8bbbdcb48c03380f2c23a1a6c9f9fc)